docs/HowTo/BreakingChanges.md

```diff
@@ -0,0 +1,142 @@
+quirking model.
+
+Runtime issues are when a new runtime is placed on a machine and the same binaries are run. If a binary was compiled for .NET 4.0,
+it will then run in .NET 4.0 compatibility mode on 4.5+. Many of the changes that affect 4.5 will not affect a binary compiled
+for 4.0. This is specific to the AppDomain and depends on the settings of the entry assembly. 
+
+Retargeting issues are those that arise when an assembly that was targeting 4.0 is now set to target 4.5. Now, it opts into many of
+the new features, with some breaking changes to old features. Again, this is dictated by the entry assembly, so the console app that
+uses the assembly, or the website that references the assembly.
+
+Sometimes, there are new behaviors that can be toggled on or off. This is often done by registry keys or configuration values, but
+depends on the specific change if those are available.
+
+This analysis is only applicable to assemblies that are designed to run on the .NET Framework 4.0+.  This is because the .NET 
+Framework is an in-place update that has compatibility modes for previous versions of the framework. This means that you can update
+to the newest version, currently 4.6, and still be able to target modes for 4.0, 4.5, 4.5.1, and 4.5.2.  This is done through a 
+mechanism called quirking and is a key reason why applications designed for v4.0 should work on later versions (although we highly
+recommend testing, especially critical applications, on the runtime you expect to be using).
+
+## Quirks
+Quirks are an important concept to understand .NET compatibility.  Quirks are special branches within the .NET Framework the outcome of which
+depends on certain factors.  These quirks are generally consistent within a single AppDomain and depends on the settings of the entry assembly.
+Quirks can come in a few different forms:
+
+- Framework version the assembly was compiled for
+- Certain configuration values
+- Certain registry settings
+
+## Target Framework Moniker
+An assembly can be compiled for a specific framework. This can help ensure that it will only run on later versions; i.e., if you compile
+against 4.5, you cannot add this to a 4.0 project.  However, you may add it to any version greater than or equal to 4.5.  This is one of the simplest
+ways to opt into a quirk, and, if compiling in VS, this is done automatically. If VS is not being used to compile the assembly, then a one line
+global assembly attribute will add it manually.
+
+A TFM is automatically added by VS projects during the compilation step. If you create a project that targets .NET Framework 4.6 and compile, you
+will find a new file located at `%temp%\.NETFramework,Version=v4.6.AssemblyAttributes.cs` in which the TFM is described:
+
+```csharp
+// <autogenerated />
+using System;
+using System.Reflection;
+[assembly: global::System.Runtime.Versioning.TargetFrameworkAttribute(".NETFramework,Version=v4.6", FrameworkDisplayName = ".NET Framework 4.6")]
+```

```

Comment:
In addition to this, MSBuild will auto-generate a TargetFrameworkAttribute if there's a TFM specified in the project file. I'm not sure if it's useful or not to include that information here.